tests/reftests/scripts.test

```diff
@@ -0,0 +1,592 @@
+  if test -n "$OPAM_SWITCH_PREFIX"
+    eval (opam env --revert --shell=fish --readonly 2> /dev/null)
+  end
+  source ${BASEDIR}/root/opam-init/variables.fish > /dev/null 2> /dev/null
+end; or true
+### cat root/opam-init/init.csh
+if ( $?prompt ) then
+  if ( -f ${BASEDIR}/root/opam-init/env_hook.csh ) source ${BASEDIR}/root/opam-init/env_hook.csh >& /dev/null
+endif
+
+if ( -f ${BASEDIR}/root/opam-init/variables.csh ) then
+    if ( ! ${?OPAM_SWITCH_PREFIX} ) then
+    setenv OPAM_SWITCH_PREFIX ""
+  endif
+  if ( "x$OPAM_SWITCH_PREFIX" != "x" ) then
+    eval `opam env --revert --shell=csh --readonly`
+  endif
+  source ${BASEDIR}/root/opam-init/variables.csh >& /dev/null
+endif
+### : Variables scripts :
+### cat root/opam-init/variables.sh
+# Prefix of the current opam switch
+OPAM_SWITCH_PREFIX='${BASEDIR}/root/fake'; export OPAM_SWITCH_PREFIX;
+# Current opam switch man dir
+MANPATH="$MANPATH":'${BASEDIR}/root/fake/man'; export MANPATH;
+# Binary dir for opam switch fake
+PATH='${BASEDIR}/root/fake/bin':"$PATH"; export PATH;
+### cat root/opam-init/variables.zsh
+cat: root/opam-init/variables.zsh: No such file or directory
+# Return code 1 #
+### cat root/opam-init/variables.fish
+# Prefix of the current opam switch
+set -gx OPAM_SWITCH_PREFIX '${BASEDIR}/root/fake';
+# Current opam switch man dir
+if [ (count $MANPATH) -gt 0 ]; set -gx MANPATH $MANPATH '${BASEDIR}/root/fake/man'; end;
+# Binary dir for opam switch fake
+set -gx PATH '${BASEDIR}/root/fake/bin' $PATH;
+### cat root/opam-init/variables.csh
+# Prefix of the current opam switch
+if ( ! ${?OPAM_SWITCH_PREFIX} ) setenv OPAM_SWITCH_PREFIX ""
+setenv OPAM_SWITCH_PREFIX '${BASEDIR}/root/fake'
+# Current opam switch man dir
+if ( ! ${?MANPATH} ) setenv MANPATH ""
+setenv MANPATH "$MANPATH":'${BASEDIR}/root/fake/man'
+# Binary dir for opam switch fake
+if ( ! ${?PATH} ) setenv PATH ""
+setenv PATH '${BASEDIR}/root/fake/bin':"$PATH"
+### : Env hook scripts :
+### cat root/opam-init/env_hook.sh
+OPAMNOENVNOTICE=true; export OPAMNOENVNOTICE;
+_opam_env_hook() {
+ local previous_exit_status=$?;
+ eval $(opam env --shell=bash --readonly 2> /dev/null <&- );
+ return $previous_exit_status;
+};
+if ! [[ "$PROMPT_COMMAND" =~ _opam_env_hook ]]; then
+    PROMPT_COMMAND="_opam_env_hook;$PROMPT_COMMAND";
+fi
+### cat root/opam-init/env_hook.zsh
+OPAMNOENVNOTICE=true; export OPAMNOENVNOTICE;
+_opam_env_hook() {
+    eval $(opam env --shell=zsh --readonly 2> /dev/null <&-);
+}
+typeset -ag precmd_functions;
+if [[ -z ${precmd_functions[(r)_opam_env_hook]} ]]; then
+    precmd_functions+=_opam_env_hook;
+fi
+### cat root/opam-init/env_hook.fish
+set -gx OPAMNOENVNOTICE true;
+function __opam_env_export_eval --on-event fish_prompt;
+    eval (opam env --shell=fish --readonly 2> /dev/null);
+end
+### cat root/opam-init/env_hook.csh
+if ( ! ${?OPAMNOENVNOTICE} ) setenv OPAMNOENVNOTICE ""
+setenv OPAMNOENVNOTICE true
+alias precmd 'eval `opam env --shell=csh --readonly`'
+### : Complete scripts :
```

Comment:
do we need to have complete scripts here??

Reply:
I'd say probably not